pavelib/quality.py

```diff
@@ -258,6 +264,16 @@ def run_quality(options):
     pep8_reports = u' '.join(pep8_files)
 
     try:
+        violations_str = None
+        with open(pep8_reports, 'r') as f:
+            violations_list = f.readlines()
+            violations_str = '\n'.join(violations_list)
+
+        if violations_str:
+            raise BuildFailure(
+                'Number of pep8 violations should be 0; found these violations:\n' + violations_str
+            )
+
         sh(
             "diff-quality --violations=pep8 {pep8_reports} {percentage_string} "
```

Comment:
ugh 15 min in the diff-quality source says that'd be a tall task, I think. I'll investigate this first. Just simply printing out the violations would be an improvement to what we have.